platform/osgi/platform-osgi-conditions/src/main/java/org/codice/ddf/condition/SignerCondition.java

```diff
@@ -0,0 +1,198 @@
+   * objects or immutable postponed conditions, and must always be called inside a permission check.
+   * Mutable postponed Condition objects will be called with the grouped version {@link
+   * #isSatisfied(Condition[],Dictionary)} at the end of the permission check.
+   *
+   * @return {@code true} to indicate the Conditions is satisfied. Otherwise, {@code false} if the
+   *     Condition is not satisfied.
+   */
+  @Override
+  public boolean isSatisfied() {
+    Map<X509Certificate, List<X509Certificate>> signerCertificates =
+        bundle.getSignerCertificates(Bundle.SIGNERS_TRUSTED);
+    Set<X509Certificate> x509Certificates = signerCertificates.keySet();
+    for (String arg : args) {
+      boolean satisfied = false;
+      for (X509Certificate x509Certificate : x509Certificates) {
+        String cn = getExtendedCertAttribute(x509Certificate.getSubjectX500Principal(), BCStyle.CN);
+        List<String> subjectAlternativeNames = getSubjectAlternativeNames(x509Certificate);
+        if (arg.equals(cn) || subjectAlternativeNames.contains(arg)) {
+          satisfied = true;
+        }
+      }
+      if (!satisfied) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  /**
+   * Returns whether the Condition is mutable. A Condition can go from mutable ({@code true}) to
+   * immutable ({@code false}) over time but never from immutable ({@code false}) to mutable ({@code
+   * true}).
+   *
+   * @return {@code true} {@link #isSatisfied()} can change. Otherwise, {@code false} if the value
+   *     returned by {@link #isSatisfied()} will not change for this condition.
+   */
+  @Override
+  public boolean isMutable() {
+    return false;
+  }
+
+  /**
+   * Returns whether the specified set of Condition objects are satisfied. Although this method is
+   * not static, it must be implemented as if it were static. All of the passed Condition objects
+   * will be of the same type and will correspond to the class type of the object on which this
+   * method is invoked. This method must be called inside a permission check only.
+   *
+   * @param conditions The array of Condition objects, which must all be of the same class and
+   *     mutable. The receiver must be one of those Condition objects.
+   * @param context A Dictionary object that implementors can use to track state. If this method is
+   *     invoked multiple times in the same permission check, the same Dictionary will be passed
+   *     multiple times. The SecurityManager treats this Dictionary as an opaque object and simply
+   *     creates an empty dictionary and passes it to subsequent invocations if multiple invocations
+   *     are needed.
+   * @return {@code true} if all the Condition objects are satisfied. Otherwise, {@code false} if
+   *     one of the Condition objects is not satisfied.
+   */
+  @Override
+  public boolean isSatisfied(Condition[] conditions, Dictionary<Object, Object> context) {
```

Comment:
Identical implementation in multiple subclass, that would be a good candidate to be defined as a default interface method

Reply:
I don't have any control over the interface. I didn't really want to make a parent class because I don't even know if this implementation is correct or correct for all of them because the spec is completely ambiguous on what to do with this. Just going to leave it for now until we figure out what to do with it or what the right approach is.